blocks/embed/embed.js

```diff
@@ -0,0 +1,63 @@
+const getDefaultEmbed = (url) => `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%">
+      <iframe src="${url.href}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute; overflow: visible;" allowfullscreen=""
+        scrolling="yes" allow="encrypted-media" title="Content from ${url.hostname}" loading="lazy">
+      </iframe>
+    </div>`;
+
+const embedOfferings = (url) => {
+  const embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding: 65% 0;">
+    <iframe src="${url.href}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute; overflow: visible;" allowfullscreen=""
+      scrolling="yes" allow="encrypted-media" title="Content from ${url.hostname}" loading="lazy">
+    </iframe>
+  </div>`;
+  return embedHTML;
+};
+
+const embedDiversity = (url) => {
+  const embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 42.75%;">
+      <iframe src="${url.href}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute; overflow: visible;" allowfullscreen=""
+        scrolling="yes" allow="encrypted-media" title="Content from ${url.hostname}" loading="lazy">
+      </iframe>
+    </div>`;
+  return embedHTML;
+};
+
+const loadEmbed = (block, link) => {
+  if (block.classList.contains('embed-is-loaded')) {
+    return;
+  }
+
+  const EMBEDS_CONFIG = [
```

Comment:
why this complicated structure? do we expect many more embeds soon? otherwise I would just do `{ 'offerings': embedOfferings, ... } `

Reply:
Agree that we can further simplify this but did not know whether there will be more Embed types that may be added in the future from external video and audio providers. Tried to follow the pattern laid out in the reference implementation that could be found at - https://github.com/adobe/helix-block-collection/tree/main/blocks/embed